platforms/enterprise/enterprise-operations/src/main/java/org/gradle/internal/configurationcache/ConfigurationCacheCheckFingerprintBuildOperationType.java

```diff
@@ -0,0 +1,155 @@
+     *
+     *     <li><b>Partially invalidated</b>: some projects are invalidated and have to be reconfigured.
+     *     The configuration is rebuilt but cached data of the unaffected projects is reused.
+     *     Corresponds to {@link CheckStatus#PARTIAL}.
+     *     The {@link #getBuildInvalidationReasons()} list is empty.
+     *     The {@link #getProjectInvalidationReasons()} list contains information about invalidated projects.</li>
+     *
+     *     <li><b>Fully invalidated</b>: some of the build-level inputs has changed. The entry is discarded and configuration phase reruns from scratch.
+     *     Corresponds to {@link CheckStatus#INVALID}.
+     *     The {@link #getBuildInvalidationReasons()} list contains the invalidation reasons.
+     *     The {@link #getProjectInvalidationReasons()} list is empty.</li>
+     * </ul>
+     *
+     * @since 8.10
+     */
+    public interface Result {
+
+        /**
+         * Returns the overall status of the fingerprint check.
+         *
+         * @return the status
+         * @since 8.10
+         */
+        CheckStatus getStatus();
+
+        /**
+         * Returns the list of the build fingerprint invalidation reasons. Can be empty if all build-level inputs are up-to-date.
+         * The entry may still be partially invalidated if any of the project-scoped fingerprints are invalid.
+         * <p>
+         * Not all invalidation reasons may be returned. However, if the entry is invalidated, at least one reason is listed.
+         * <p>
+         * The first invalidation in the list is what Gradle shows to the user.
+         *
+         * @return the list of invalidation reasons
+         * @since 8.10
+         */
+        List<FingerprintInvalidationReason> getBuildInvalidationReasons();
+
+        /**
+         * Returns a list of project-level invalidation reasons. Only contains entries for projects that were invalidated.
+         * This list is empty if the whole build is invalidated, i.e. when {@link #getBuildInvalidationReasons()} returns a non-empty list.
+         * <p>
+         * All invalidated project have entries in this list, however, not all invalidation reasons may be listed.
+         * <p>
+         * The first invalidation of the first project in the list is what Gradle shows to the user.
+         *
+         * @return a list of invalidation projects with their invalidation reasons
+         * @since 8.10
+         */
+        List<ProjectInvalidationReasons> getProjectInvalidationReasons();
+    }
+
+    /**
+     * Contains data of a single invalidated project.
+     *
+     * @since 8.10
+     */
+    public interface ProjectInvalidationReasons {
+        /**
+         * Returns the identity path of the invalidated project.
+         *
+         * @return the identity path
+         * @since 8.10
+         */
+        String getIdentityPath();
```

Comment:
🤔  Here are two other options to achieve the same results. I am not sure which one of all of them is better.

- Use `getBuildTreePath()` as we started using in all new APIs, including the Tooling API
- Split this method into `getBuildPath()` and `getProjectPath()` as it is done in many other build operations

@wolfs, do you have an opinion here? I recall there were difficulties in various places with splitting up the identity path.

Reply:
>I recall there were difficulties in various places with splitting up the identity path.

Here I just used what was available in the underlying CC model, but I'm open to change this if we need to provide the path as two parts.

Reply:
> * Use getBuildTreePath() as we started using in all new APIs, including the Tooling API

Isn't `buildTreePath` the same as `identityPath`, though? On `ProjectInternal` one is `String`, another is `Path`, but that looks like the only difference.

Reply:
>do you have an opinion here? I recall there were difficulties in various places with splitting up the identity path.

❓ @wolfs, @sorin-florea what would be the conclusion here? Should we split tree path into build and project path?

Reply:
Let's split it and use `buildPath`/`projectPath`.